library/network_connections.py

```diff
@@ -676,7 +676,14 @@ def setting_ip_config_get_routes(self, s_ip):
     def connection_ensure_setting(self, connection, setting_type):
         setting = connection.get_setting(setting_type)
         if not setting:
-            setting = setting_type.new()
+            util_nm = Util.NM()
+            if setting_type is getattr(Util.NM(), "SettingMatch", None):
```

Comment:
This will not doe the right thing if `setting_type` is `None`. Not sure if this can happen.

Reply:
I will switch back to the older version then:
`if hasattr(util_nm, 'SettingMatch') and setting_type is util_nm.SettingMatch:`

Reply:
`setting_type` must not be `None` here, so I don't think that is any concern.

There was no problem with the last version (except that `util_nm` was unnecessary).